sympy/solvers/solveset.py

```diff
@@ -209,6 +209,32 @@ def _invert_complex(f, g_ys, symbol):
                                                log(Abs(g_y))), S.Integers)
                                for g_y in g_ys])
             return _invert_complex(f.args[0], exp_invs, symbol)
+
+    if isinstance(f, (sin, csc)):
+        n = Dummy('n')
+        if isinstance(g_ys, FiniteSet):
+            fi = sympify('a' + f.func.__name__)
+            sin_csc_invs = Union(*[imageset(Lambda(n, n*pi + (-1)**n*fi(g_y)),
```

Comment:
We can use the other formula for this as well but this is better because it represent the solution `not` in the form of union. Hence a simplified form.